command/agent/command.go

```diff
@@ -621,6 +621,9 @@ func (c *Command) setupTelemetry(config *Config) error {
 
 	metricsConf := metrics.DefaultConfig("nomad")
 	metricsConf.EnableHostname = !telConfig.DisableHostname
+	if telConfig.UseName {
```

Comment:
Can we just detect if UseName is set and set the appropriate EnableHostname. In the docs we should note if set it will override `DisableHostname`